x-pack/plugins/enterprise_search/public/applications/app_search/index.tsx

```diff
@@ -58,18 +62,12 @@ export const AppSearchUnconfigured: React.FC = () => (
   </Switch>
 );
 
-export const AppSearchConfigured: React.FC<InitialAppData> = (props) => {
-  const { initializeAppData } = useActions(AppLogic);
+export const AppSearchConfigured: React.FC<Required<InitialAppData>> = (props) => {
   const {
-    hasInitialized,
     myRole: { canManageEngines, canManageMetaEngines },
-  } = useValues(AppLogic);
+  } = useValues(AppLogic(props));
```

Comment:
TIL there is [another way](https://kea.js.org/blog/kea-2.3) to do this. Definitely think what we have is the right solution but it's nice to see that we could do something else in a component, if needed. Particularly if we needed shared props in multiple child components

Reply:
TIL!!! Thanks Scotty, that's incredibly helpful if we start diving into keyed logic more!

Reply:
Yeah, I'm beginning the process of migrating App Search's Role Mappings and see a lot of shared code in our logic files. Thinking of either creating a shared logic file or finding a way to inherit reducers and actions from some sort of shared parent logic file.